hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/bucket/BucketCache.java

```diff
@@ -560,6 +564,21 @@ protected void cacheBlockWithWaitInternal(BlockCacheKey cacheKey, Cacheable cach
     }
   }
 
+  public BucketEntry getBlockForReference(BlockCacheKey key) {
```

Comment:
Mind explaining a bit what is the usage for this method? And add some comments? Does this method needs to be public?

Reply:
Thanks, I have added javadoc comments now. This method "resolves" the reference to the referred file and looks for a block from the referred file in the cache. The idea is to allow references to "reuse" the blocks from the referred files if those are already in the cache.